106-self-healing-feature-in-operator.md

```diff
@@ -0,0 +1,295 @@
+The notifier class returns the `action` that is going to be taken on the flagged anomaly.
+These actions have three types:
+* FIX - Start the anomaly fix
+* CHECK - Delay the anomaly fix
+* IGNORE - Ignore the anomaly fix
+
+The default `NoopNotifer` always sets the notifier action as `IGNORE`, which  means that the detected anomaly will be silently ignored and no notification is sent to the user.
+
+Cruise Control also provides [custom notifiers](https://github.com/linkedin/cruise-control/wiki/Configure-notifications) like Slack Notifier, Alerta Notifier etc. for notifying users regarding the anomalies. There are multiple other [self-healing notifier](https://github.com/linkedin/cruise-control/wiki/Configurations#selfhealingnotifier-configurations) related configurations you can use to make notifiers more efficient as per the use case.
+
+#### Self Healing
+
+If self-healing is enabled, then an action is returned by the notifier to would decide whether the anomaly should be fixed or not.
+If the notifier has returned `FIX` as the action then the classes which are responsible for resolving the anomaly would be called.
+Each detectable anomaly is handled by a specific detector class which then uses another remediation class to run a fix.
+For example, the `GoalViolations` class uses the `RebalanceRunnable` class, the `DiskFailure` class use the `RemoveDisksRunnable` class and so on.
+An optimization proposal will then be generated by these `Runnable` classes and that proposal will be applied on the cluster to fix the anomaly.
+In case the anomaly detected is unfixable for e.g. violated hard goals that cannot be fixed typically due to lack of physical hardware (insufficient number of racks to satisfy rack awareness, insufficient number of brokers to satisfy Replica Capacity Goal, or insufficient number of resources to satisfy resource capacity goals), the anomaly wouldn't be fixed and the Cruise Control logs will be updated with `self-healing is not possible due to unfixable goals` warning.
+
+### Proposal
+
+This proposal allows the users to have their cluster balanced automatically whenever an anomaly is detected by Cruise Control.
+To ensure that the operator drives the rebalance we will be leveraging the Cruise Control self-healing mechanism but not completely.
+When using self-healing in Cruise Control, the rebalance are triggered automatically in the cluster which means that the operator wouldn't have the information about when a rebalance is happening.
+To resolve this issue, we will only make use of the ability of Cruise Control to detect the anomalies and based on the detection, we will then notify the operator to run the rebalance.
+Doing this will provide us with following merits:
+* ensure that the operator know what is going on in the cluster.
+* better debugging.
+* to drive the rebalances/fixes from the operator.
+
+#### `Full` mode in auto-rebalancing feature of Strimzi
```

Comment:
I guess it is `full` mode in lowercase as that is how it will be in the custom resource? (not just here but also in the other places)